provider/docker/builder_test.go

```diff
@@ -0,0 +1,69 @@
+package docker
+
+import (
+	docker "github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/container"
+	"github.com/docker/engine-api/types/network"
+	"github.com/docker/go-connections/nat"
+)
+
+func containerJSON(ops ...func(*docker.ContainerJSON)) docker.ContainerJSON {
+	c := &docker.ContainerJSON{
+		ContainerJSONBase: &docker.ContainerJSONBase{
+			Name:       "fake",
+			HostConfig: &container.HostConfig{},
+		},
+		Config: &container.Config{},
+		NetworkSettings: &docker.NetworkSettings{
+			NetworkSettingsBase: docker.NetworkSettingsBase{},
+		},
+	}
+
+	for _, op := range ops {
+		op(c)
+	}
+
+	return *c
```

Comment:
How about returning a pointer since we are already applying the ops on the pointer?